src/sentry/migrations/0242_prefill_callsigns.py

```diff
@@ -0,0 +1,623 @@
+# -*- coding: utf-8 -*-
+import re
+from itertools import count
+from contextlib import contextmanager
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models, transaction
+from sentry.utils.strings import tokens_from_name, iter_callsign_choices
+
+
+class RollbackLocally(Exception):
+    pass
+
+
+@contextmanager
+def catchable_atomic():
+    try:
+        with transaction.atomic():
+            yield
+    except RollbackLocally:
+        pass
+
+
+def get_callsigns(projects):
+    rv = {}
+
+    for project in projects:
+        if project.callsign is not None:
+            rv[project.callsign] = project.id
+            continue
+        for callsign in iter_callsign_choices(project.name,
+                                              team_name=project.team.name):
+            if callsign in rv:
+                continue
+            rv[callsign] = project.id
+            break
+
+    return dict((v, k) for k, v in rv.iteritems())
+
+
+class Migration(DataMigration):
+
+    def forwards(self, orm):
+        from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+        from sentry.models.counter import increment_project_counter
+
+        Organization = orm['sentry.Organization']
+        Group = orm['sentry.Group']
+        Project = orm['sentry.Project']
+        ProjectOption = orm['sentry.ProjectOption']
+
+        queryset = Organization.objects.all()
+
+        for org in RangeQuerySetWrapperWithProgressBar(queryset):
+            cnt = orm['sentry.Counter'].objects.all()
```

Comment:
What's this query for? It seems wasteful and it'd be repeated on each iteration while the result isn't being used.